sdk/resource/resource.go

```diff
@@ -51,29 +74,71 @@ func Merge(a, b *Resource) *Resource {
 	if b == nil {
 		return a
 	}
-	res := &Resource{
-		labels: map[core.Key]core.Value{},
+
+	n := len(a.sorted)
+	if len(b.sorted) > len(a.sorted) {
+		n = len(b.sorted)
+	}
+	// At a minimum the merge will be as large as the largest resource.
+	s := make([]core.KeyValue, 0, n)
+	k := make(map[core.Key]struct{}, n)
+	ai, bi := 0, 0
+	for ; ai < len(a.sorted) && bi < len(b.sorted); ai, bi = ai+1, bi+1 {
+		akv := a.sorted[ai]
+		s = append(s, akv)
```

Comment:
A conventional merge sort would work here without the call to `sort.Slice()`.  I wrote one up (about the same length, 30 lines) and it's between 20% faster and 100% faster, depending on the number of keys in the resources.

But this is relatively easier to understand, and as you say, almost certainly not anywhere near the hot path.

Reply:
I think my goal for this PR at this point is to simplify and get the method signatures to what we want. I'll add a note here though about potentially optimizing this in the future.